pkg/k8shandler/secret.go

```diff
@@ -92,3 +94,33 @@ func (clusterRequest *ClusterLoggingRequest) RemoveSecret(secretName string) err
 
 	return nil
 }
+
+func calcSecretHashValue(secret *core.Secret) (string, error) {
+	hashValue := ""
+	var err error
+
+	if secret == nil {
+		return hashValue, nil
+	}
+
+	hashKeys := []string{}
+	rawbytes := []byte{}
+
+	// we just want the keys here to sort them for consistently calculated hashes
+	for key := range secret.Data {
+		hashKeys = append(hashKeys, key)
+	}
+
+	sort.Strings(hashKeys)
+
+	for _, key := range hashKeys {
+		rawbytes = append(rawbytes, secret.Data[key]...)
+	}
+
+	hashValue, err = utils.CalculateMD5Hash(string(rawbytes))
```

Comment:
is there a need to calculate the hash of the keys and the content.  Dont we need to be concerned with the key names changing?

Reply:
we don't need to worry about the keys changing because the operator ensures that it will be the correct key name. we need to know if the content (certs/keys) changed to know if we need to restart kibana